apps/server/src/shared/infra/identity-management/keycloak-identity-management.spec.ts

```diff
@@ -0,0 +1,255 @@
+							resetPassword: jest.fn(),
+							del: jest.fn(),
+						},
+					},
+				},
+				{
+					provide: 'KeycloakSettings',
+					useValue: 'empty',
+				},
+			],
+		}).compile();
+		idm = module.get<IIdentityManagement>('IdentityManagement');
+		kcAdminClient = module.get<KeycloakAdminClient>('KeycloakAdminClient');
+	});
+
+	it('should be defined', () => {
+		expect(idm).toBeDefined();
+	});
+
+	describe('createAccount', () => {
+		it('should allow to create an account', async () => {
+			const accountId = 'user-1-id';
+			const createUserMock = jest.spyOn(kcAdminClient.users, 'create').mockResolvedValue({ id: accountId });
+			const resetPasswordMock = jest.spyOn(kcAdminClient.users, 'resetPassword');
+			const testAccount = { userName: 'test', email: 'test@test.test' };
+			const testAccountPassword = 'test';
+
+			const ret = await idm.createAccount(testAccount, testAccountPassword);
+
+			expect(ret).not.toBeNull();
+			expect(ret).toBe(accountId);
+			expect(createUserMock).toBeCalledWith(
+				expect.objectContaining({ username: testAccount.userName, email: testAccount.email })
+			);
+			expect(resetPasswordMock).toBeCalledWith(
+				expect.objectContaining({
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+					credential: expect.objectContaining({ value: testAccountPassword }),
+				})
+			);
+		});
+
+		it('should reject if account create fails', async () => {
+			jest.spyOn(kcAdminClient.users, 'create').mockResolvedValueOnce({ id: 'accountId' }).mockRejectedValue('error');
+			const resetPasswordMock = jest.spyOn(kcAdminClient.users, 'resetPassword').mockRejectedValue('error');
+			const deleteAccountMock = jest.spyOn(kcAdminClient.users, 'del');
+
+			const testAccount = { userName: 'test', email: 'test@test.test' };
+			const testAccountPassword = 'test';
+
+			await expect(idm.createAccount(testAccount, testAccountPassword)).rejects.toBeTruthy();
+			expect(resetPasswordMock).toHaveBeenCalled();
+			expect(deleteAccountMock).toHaveBeenCalled();
+			resetPasswordMock.mockClear();
+			deleteAccountMock.mockClear();
+
+			await expect(idm.createAccount(testAccount, testAccountPassword)).rejects.toBeTruthy();
+			expect(resetPasswordMock).not.toHaveBeenCalled();
+			expect(deleteAccountMock).not.toHaveBeenCalled();
+		});
+	});
+
+	describe('findAccountById', () => {
+		it('should find an existing account', async () => {
+			const accountId = 'user-1-id';
+			const accountUserName = 'user-1';
+			const accountEmail = 'user@mail';
+			const accountFirstName = 'user';
+			const accountLastName = '1';
+
+			jest.spyOn(kcAdminClient.users, 'findOne').mockResolvedValue({
+				id: accountId,
+				username: accountUserName,
+				email: accountEmail,
+				firstName: accountFirstName,
+				lastName: accountLastName,
+			});
+
+			const ret = await idm.findAccountById(accountId);
+
+			expect(ret).not.toBeNull();
+			expect(ret).toEqual(
+				expect.objectContaining({
+					id: accountId,
```

Comment:
This is fine. A quick alternative if you need two objects that are the same, but are not the same reference, is using the spread operator `const objectCopy = { ...originalObject }`
It creates a flat copy.

Reply:
These objects are different (internal Keycloak representation and IAccount transfer object). They just have a similar fields 😄